jobfunnel/config/cli.py

```diff
@@ -0,0 +1,393 @@
+        help='Select a function to calculate delay times with.'
+    )
+
+    return parser.parse_args()
+
+
+def config_builder(args: argparse.Namespace) -> JobFunnelConfigManager:
+    """Parse the JobFunnel configuration settings into a JobFunnelConfigManager.
+
+        args [argparse.Namespace]: cli arguments from argparser
+    """
+    # Init and pop args that are cli-only and not in our schema
+    args_dict = vars(args)
+    settings_yaml_file = args_dict.pop('settings_yaml_file')
+    output_folder = args_dict.pop('output_folder')
+    args_dict.pop('do_recovery_mode')  # NOTE: this is handled in __main__
+
+    # Load config dict from the YAML if passed
+    config = {'search': {}, 'delay': {}, 'proxy': {}}
+    if settings_yaml_file:
+        config.update(
+            yaml.load(open(settings_yaml_file, 'r'), Loader=yaml.FullLoader)
+        )
+
+        # TODO: need a way to handle injecting defaults with YAML but also
+        # without an incomplete set of arguments here. Cerberus should do this
+        # if we further break down config into sub-configs.
+        missing_attrs = []  # type: List[str]
+        for attr in ['master_csv_file', 'block_list_file', 'cache_folder',
+                     'duplicates_list_file']:
+            if attr not in config:
+                missing_attrs.append(attr)
+        if missing_attrs:
+            raise ValueError(
+                f"Passed YAML {settings_yaml_file} fields are missing or "
+                f"invalid: {missing_attrs}"
+            )
+
+    # Handle output_folder argument which is a shortcut to specifying all paths
+    user_passed_paths = bool(
+        (
+            args_dict['master_csv_file'] and args_dict['block_list_file'] and
+            args_dict['duplicates_list_file'] and args_dict['cache_folder']
+        ) or (
+            config.get('master_csv_file') and config.get('block_list_file') and
+            config.get('duplicates_list_file') and config.get('cache_folder')
+        )
+    )
+    if output_folder == DEFAULT_OUTPUT_DIRECTORY and not user_passed_paths:
+
+        # We are using all defaults only (no -s or paths passed)
+        config['master_csv_file'] = DEFAULT_MASTER_CSV_FILE
+        config['block_list_file'] = DEFAULT_BLOCK_LIST_FILE
+        config['duplicates_list_file'] = DEFAULT_DUPLICATES_FILE
+        config['cache_folder'] = DEFAULT_CACHE_DIRECTORY
+
+    elif output_folder != DEFAULT_OUTPUT_DIRECTORY and user_passed_paths:
+
+        # User cannot specify both output folder and other paths
+        raise ValueError(
+            "When providing output_folder, do not also provide -csv, -blf"
+            ", -dlf,  -cache or -s, as paths are defined by the output folder."
+            " If specifying file paths you must pass all the arguments and "
+            "not pass -o."
+        )
+
+    # Inject any cli, non-default attributes (excluding paths)
+    for key, value in args_dict.items():
```

Comment:
This is overwriting options in the `settings.yaml` file, even if the user doesn't intend to do so. For example, if you specified `USA_ENGLISH` as your locale in the settings file, JobFunnel still uses the `CANADA_ENGLISH` scraper, because that is the default that gets filled in by the parser when the user doesn't pass any locale to the command line.

Reply:
oops, I wanted to be rid of the defaults checking, but I think I'll need to add it back, to ensure we only inject non-defaults.

Reply:
Ok this should be fixed now, feel free to give it a go to confirm.